### DISCUSSION TITLE
Nuxt3: using 404.vue and [slug].vue in /pages

### DISCUSSION THREAD

**Discussion by davidfrgla** [nuxt/nuxt#17168] [Questions]
Hi everyone,

I am trying to setup a file structure like that:
```
- pages/
---- index.vue
---- [slug].vue
---- 404.vue
```

When starting the app, it doesn't recognise the 404 page and goes to [slug] with `$route.params.slug` set to 404

Any solution ?

**Comment by raducretu** [comment]
+1

**Comment by alpeshvmodi** [comment]
+1

**Comment by mtskf** [comment]
You can put 404.vue or error.vue in the root folder.
https://v3.nuxtjs.org/guide/features/error-handling/#rendering-an-error-page
```
- yourproject/
---- pages/
------- index.vue
------- [slug].vue
---- error.vue
```

**Comment by izzudd** [comment]
I solve this with nesting `index.vue` inside `404` folder like this.

```
.
└── pages
    ├── 404
    │   └── index.vue
    ├── [slug].vue
    └── index.vue
```

That works for me.

**Comment by wernersa** [comment]
For anybody looking for a solution to render 404 for SSG - Add this to nuxt.config.js

```ts
  nitro: {
    prerender: {
      routes: ['/404.html'] // Hacky way of rendering a static 404 error from the [slug].vue
    }
  },
```
